main/src/ca/uwaterloo/flix/language/ast/OccurrenceAst.scala

```diff
@@ -27,7 +27,7 @@ object OccurrenceAst {
                   reachable: Set[Symbol.DefnSym],
                   sources: Map[Source, SourceLocation])
 
-  case class Def(ann: Ast.Annotations, mod: Ast.Modifiers, sym: Symbol.DefnSym, cparams: List[OccurrenceAst.FormalParam], fparams: List[OccurrenceAst.FormalParam], exp: OccurrenceAst.Expression, context: DefContext, tpe: MonoType, purity: Purity, loc: SourceLocation)
+  case class Def(ann: Ast.Annotations, mod: Ast.Modifiers, sym: Symbol.DefnSym, cparams: List[(OccurrenceAst.FormalParam, Occur)], fparams: List[(OccurrenceAst.FormalParam, Occur)], exp: OccurrenceAst.Expression, context: DefContext, tpe: MonoType, purity: Purity, loc: SourceLocation)
```

Comment:
Ocurence information is in a tuple because not all formal parameters have occurrence information. effect operations don't have implementations so the field would be meaningless there.

Reply:
I see. Would it make sense to make it optional on FormalParam then, or would that make everything worse?

Reply:
Yeah i tried it out but formal param is used in effects and trycatch handlers etc - makes too much of a mess in my opinion